.github/workflows/pr-dotnet.yml

```diff
@@ -23,3 +23,12 @@ jobs:
       - name: Lint .NET Solution
         working-directory: BenchPress
         run: dotnet format --exclude ./bicep/ --verify-no-changes
+      - name: Run Unit Tests
```

Comment:
This should probably be a different job from "Lint .NET Solution". A new job specifically for running unit tests should be created. 

Also, the solution will need to be built before testing.

Reply:
or we can simply rename the job. Jobs execute on different runners so from a performance point of view linting and running unit tests in the same job does offer a slight performance boost.

Reply:
Fair point. I'd still be a fan of separate jobs from a separation of concerns and organizational perspective. The steps involved in each job (term used just for sake of discussion) aren't dependent on each other and are unrelated. It's only my opinion, but the optimization into a single job doesn't compensate for breaking the separation of concerns for each job.

Just by $0.02, I'll defer to @hattan and @polatengin for the final recommendation.

Reply:
> For the pr-dotnet.yml workflow, consider renaming the root job

For pr-powershell.yml, we do split up into linting and unit tests; for consistency sake, should I change that too to just one big job vs 2? @marclerwick @hattan @polatengin  

I'm okay with one job for performance sake, but would prefer to keep them separated for readability.

Reply:
@marclerwick - I see the separation of concerns aspect, but there is a dependency somewhat. I definitely think that one job is the way to go. Aside from speed, you will be running tests against another compile since each job is on a different runner. By using the same job, the lint, build and unit-test is exactly the same build output.

Reply:
After discussing this with the team, we decided to go with two jobs.